src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+"""Module for analysis of EC-MS measurements with impulse response and
+deconvolution of mass transport effects to obtain partial current densities."""
+
+from .ec_ms import ECMSMeasurement
+
+
+
+class DecoMeasurement(ECMSMeasurement):
+    """Class implementing deconvolution of EC-MS data"""
+
+    def __intit__(self, name, **kwargs):
+        """initialize a deconvolution EC-MS measurement
+
+        Args:
+            name (str): The name of the measurement"""
+        super().__init__(name, **kwargs)
+
+    def get_partial_current(self,
+        signal_name,
+        kernel_obj,
+        tspan=None,
+        t_bg=None,
+        snr=10):
+        """Returns the deconvoluted partial current for a given signal
+
+        Args:
+            signal_name (str): Name of signal for which deconvolution is to
+                be carried out.
+            kernel_obj (Kernel): Kernel object which contains the mass transport
+                parameters
+            tspan (list): Timespan for which the partial current is returned.
+            t_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+            """
+
+
+        t_sig, v_sig = self.get_calib_signal(signal_name,
+            tspan=tspan,
+            t_bg=t_bg)
+
+        kernel = kernel_obj.generate_kernel(dt=t_sig[1]-t_sig[0])
+        kernel = np.hstack((kernel, np.zeros(len(sig) - len(kernel))))
```

Comment:
It looks here like kernel is calculated, mutated, used to make H, and neither are saved anywhere. 
I would have both the kernel values and H stored in the `kernel_obj` and accessed as properties. As mentioned in the comments to the `Kernel` class, the code would be more intuitive and powerful if you put more functionality into that class.

Reply:
Can you add a TODO here as well about storing these things?
Sorry to be annoying, but I don't think we'll look at these PR comments again so if it's not in the code it's nowhere.